docs/authentication.md

```diff
@@ -101,31 +107,14 @@ class NasaEarthdataCredentialProvider:
             password: Password to NASA Earthdata.
 
         """
-        self.username = username
-        self.password = password
+        self.session = requests.Session()
+        self.session.auth = (username, password)
 
     def __call__(self) -> S3Credential:
-        login_resp = requests.get(CREDENTIALS_API, allow_redirects=False)
-        login_resp.raise_for_status()
-
-        encoded_auth = base64.b64encode(
-            f"{self.username}:{self.password}".encode("ascii"),
-        )
-        auth_redirect = requests.post(
-            login_resp.headers["location"],
-            data={"credentials": encoded_auth},
-            headers={"Origin": CREDENTIALS_API},
-            allow_redirects=False,
-        )
-        auth_redirect.raise_for_status()
-
-        final = requests.get(auth_redirect.headers["location"], allow_redirects=False)
-        results = requests.get(
-            CREDENTIALS_API, cookies={"accessToken": final.cookies["accessToken"]}
-        )
-        results.raise_for_status()
-
-        creds = results.json()
+        """Request updated credentials."""
+        resp = self.session.get(CREDENTIALS_API, allow_redirects=True, timeout=15)
+        auth_resp = self.session.get(resp.url, allow_redirects=True, timeout=15)
+        creds = auth_resp.json()
```

Comment:
I don't think we should be using this as an example for 2 reasons:

1. The session is never closed
2. It is insecure because you're leaking EDL creds to the CREDENTIALS_API. The EDL creds should be supplied _only_ to the URS (Earthdata Login) URL.

Although your original implementation is more verbose, it is secure, and does not leave dangling resources. (There's no strong motivation for using a session since auth occurs no more than once an hour, or more likely, only once it total for most use cases.)

Reply:
The original implementation _didn't work_, see https://github.com/developmentseed/obstore/issues/271#issue-2854852119. I'm not sure why it didn't work; the current implementation was suggested by Aimee in #271, and appears to be what the [`earthaccess`](https://github.com/nsidc/earthaccess) Python package is using. 

Using a session seemed to make it easier to automatically manage cookies between the two requests, but it's not a requirement for me. In the async case I add a `close()` method to the credential provider and document that it should be called after all obstore use has finished.

Perhaps this isn't a great use for an example, and we should find something simpler for the docs.